src/midi/midi_fluidsynth.cpp

```diff
@@ -446,6 +449,74 @@ void MidiHandlerFluidsynth::Render()
 	}
 }
 
+std::string format_sf2_line(const std::string &path, const std::string &name)
+{
+	const size_t term_width = real_readw(BIOSMEM_SEG, BIOSMEM_NB_COLS);
+	assert(term_width > 0);
+	std::vector<char> line_buf(term_width);
+	snprintf(line_buf.data(), term_width, "  %-16s - %s%s\n", name.c_str(),
+	         path.c_str(), name.c_str());
+	std::string line = line_buf.data();
+
+	// If line ends with a newline, then the whole description fits into a
+	// single line - no further formatting is necessary.
+	if (line.back() == '\n') 
+		return line;
+
+	// The description was too long and got trimmed; place three dots in
+	// the end to make it clear to the user.
+	assert(line.size() > 4);
+	auto it = line.end();
+	assert(*it == '\0');
+	*(--it) = '\n';
+	*(--it) = '.';
+	*(--it) = '.';
+	*(--it) = '.';
```

Comment:
Coverity caught this:

``` text
466             // The description was too long and got trimmed; place three dots in
467             // the end to make it clear to the user.
468             assert(line.size() > 4);
469             auto it = line.end();
>>>     CID 329626:  API usage errors  (INVALIDATE_ITERATOR)
>>>     Dereferencing iterator "it" though it is already past the end of its container.
470             assert(*it == '\0');
471             *(--it) = '\n';
472             *(--it) = '.';
473             *(--it) = '.';
474             *(--it) = '.';
475             return line;
```

Reply:
This might do:
``` c++
	const std::string cutoff("...\n");
	assert(line.size() > cutoff.size())
	line.replace(line.end() - cutoff.size(), line.end(), cutoff);
```

Reply:
Oh, it's much better - thanks! Committed as: a5b85e267f0228388774963ba206219b0a3a8539